eclipse/src/saros/Saros.java

```diff
@@ -260,29 +268,34 @@ private void stopLifeCycle() {
   }
 
   private void setupLoggers() {
-    /*
-     * HACK this is not the way OSGi works but it currently fulfill its
-     * purpose
-     */
-    final ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
 
     final boolean isDebugMode = Boolean.getBoolean("saros.debug") || isDebugging(); // $NON-NLS-1$
 
-    final String log4jPropertyFile =
-        isDebugMode
-            ? "saros_debug.log4j.properties" //$NON-NLS-1$
-            : "saros_release.log4j.properties"; //$NON-NLS-1$
-
     try {
-      // change the context class loader so Log4J will find the appenders
-      Thread.currentThread().setContextClassLoader(Saros.class.getClassLoader());
+      File mainLogDir = getStateLocation().toFile();
+      final String logDir = mainLogDir + File.separator + "log";
+
+      final Level logLevel = isDebugMode ? Level.ALL : Level.WARN;
 
-      PropertyConfigurator.configure(Saros.class.getClassLoader().getResource(log4jPropertyFile));
+      URL configFileU = getBundle().getResource(LOG4J2_CONFIG_FILENAME);
```

Comment:
configFileU more readable than configFileURL ?

Reply:
No, seems like I lost the RL somewhere.